lib/mozart_fetcher/decoder.ex

```diff
@@ -0,0 +1,32 @@
+defmodule MozartFetcher.Decoder do
+  def data_to_struct(data, as: a_struct) do
+    case Jason.decode(data) do
+      {:ok, decoded} -> {:ok, to_struct(decoded, as: a_struct)}
+      {:error, _}    -> {:error}
+    end
+  end
+
+  def list_to_struct_list(data, struct) do
+    case Jason.decode(data) do
+      {:ok, data} ->
+        {:ok, Enum.map(data["components"], fn x -> to_struct(x, as: struct) end)}
+      {:error, _} -> {:error}
+    end
+
+    # build list while iterating through each section of the JSON and turning it into a struct
```

Comment:
Is the list built above? If so, can we put the comments before the code they are commenting.

Reply:
Yea I forgot to remove the comments, do you think its useful to keep them and move it up?

Reply:
Not if you were going to remove them.